azurelinuxagent/ga/exthandlers.py

```diff
@@ -524,10 +530,45 @@ def report_ext_handlers_status(self):
 
             self.report_status_error_state.reset()
 
+        self.write_ext_handlers_status_to_info_file(vm_status)
+
+    def write_ext_handlers_status_to_info_file(self, vm_status):
+        status_path = os.path.join(conf.get_lib_dir(), AGENT_STATUS_FILE)
+
+        agent_details = {
+            "AGENT_NAME": AGENT_NAME,
+            "CURRENT_VERSION": str(CURRENT_VERSION),
+            "GOAL_STATE_AGENT_VERSION": str(GOAL_STATE_AGENT_VERSION),
+            "TIMESTAMP": time.strftime("%Y-%m-%dT%H:%M:%SZ", time.gmtime())
+        }
+
+        # Convert VMStatus class to Dict.
+        data = get_properties(vm_status)
+
+        # The above class contains vmAgent.extensionHandlers
+        # (more info: azurelinuxagent.common.protocol.restapi.VMAgentStatus)
+        handler_statuses = data['vmAgent']['extensionHandlers']
+        for handler_status in handler_statuses:
+            try:
+                handler_status.pop('code', None)
+                handler_status.pop('message', None)
+                handler_status.pop('extensions', None)
+            except KeyError:
+                pass
+
+        agent_details['HANDLER_STATUS'] = handler_statuses
+        agent_details_json = json.dumps(agent_details)
+
+        digest = hash_strings(agent_details_json)
+        if digest != self.last_status_file_hash:
```

Comment:
since we are just writing to a local file I don't think we need to check for changes.

also, the time of the latest update (instead of the time of the latest update where there were changes) would be more useful, I think.